addons/mail/static/src/core/public_web/thread_model_patch.js

```diff
@@ -21,7 +21,7 @@ patch(Thread.prototype, {
         const channel_notifications =
             this.custom_notifications || this.store.settings.channel_notifications;
         if (
-            !this.mute_until_dt &&
+            !this.selfMember?.mute_until_dt &&
```

Comment:
This is true if `this.selfMember` is `undefined`. We might want to check for `this.selfMember` first.

Reply:
It's a really good idea to double check `undefined` behavior when making this kind of change, but I believe in this case not having `this.selfMember` should have been considered as valid as we still want to notify of the message for non-members. The restriction here was truly only on being non-muted, not the existence of the member.